python/ray/llm/_internal/batch/processor/sglang_engine_proc.py

```diff
@@ -0,0 +1,215 @@
+        description="The maximum number of pending requests. If not specified, "
+        "will use the default value from the SGLang engine.",
+    )
+    max_concurrent_batches: int = Field(
+        default=4,
+        description="The maximum number of concurrent batches in the engine. "
+        "This is to overlap the batch processing to avoid the tail latency of "
+        "each batch. The default value may not be optimal when the batch size "
+        "or the batch processing latency is too small, but it should be good "
+        "enough for batch size >= 64.",
+    )
+
+    # Processor stage configurations.
+    apply_chat_template: bool = Field(
+        default=True, description="Whether to apply chat template."
+    )
+    chat_template: Optional[str] = Field(
+        default=None,
+        description="The chat template to use. This is usually not needed if the "
+        "model checkpoint already contains the chat template.",
+    )
+    tokenize: bool = Field(
+        default=True,
+        description="Whether to tokenize the input before passing it to the "
+        "SGLang engine. If not, SGLang will tokenize the prompt in the engine.",
+    )
+    detokenize: bool = Field(
+        default=True,
+        description="Whether to detokenize the output.",
+    )
+
+    @root_validator(pre=True)
+    def validate_task_type(cls, values):
+        task_type_str = values.get("task_type", "generate")
+        values["task_type"] = SGLangTaskType(task_type_str)
+        return values
+
+
+def build_sglang_engine_processor(
+    config: SGLangEngineProcessorConfig,
+    preprocess: Optional[UserDefinedFunction] = None,
+    postprocess: Optional[UserDefinedFunction] = None,
+    telemetry_agent: Optional[TelemetryAgent] = None,
+) -> Processor:
+    """Construct a Processor and configure stages.
+    Args:
+        config: The configuration for the processor.
+        preprocess: An optional lambda function that takes a row (dict) as input
+            and returns a preprocessed row (dict). The output row must contain the
+            required fields for the following processing stages.
+        postprocess: An optional lambda function that takes a row (dict) as input
+            and returns a postprocessed row (dict).
+        telemetry_agent: An optional telemetry agent for collecting usage telemetry.
+
+    Returns:
+        The constructed processor.
+    """
+    ray.init(runtime_env=config.runtime_env, ignore_reinit_error=True)
+
+    stages = []
+
+    if config.apply_chat_template:
+        stages.append(
+            ChatTemplateStage(
+                fn_constructor_kwargs=dict(
+                    model=config.model_source,
+                    chat_template=config.chat_template,
+                ),
+                map_batches_kwargs=dict(
+                    zero_copy_batch=True,
+                    concurrency=(1, config.concurrency),
+                    batch_size=config.batch_size,
+                ),
```

Comment:
Please pass runtime env to all stages as the fix in #51508

Reply:
done